src/App.tsx

```diff
@@ -14,11 +14,174 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  goods: string[],
+  listIsVisible: boolean,
+  lastFunction: string,
+  inputValue: number,
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    listIsVisible: true,
+    lastFunction: '',
+    inputValue: 1,
+  };
+
+  reverseList = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].reverse(),
+      lastFunction: 'reverse',
+    }));
+  };
+
+  sortAlphabetically = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].sort((a, b) => a.localeCompare(b)),
+      lastFunction: 'sortAlpha',
+    }));
+  };
+
+  sortByLength = () => {
+    this.setState(prevState => ({
+      goods: [...prevState.goods].sort((a, b) => a.length - b.length),
+      lastFunction: 'sortByLength',
+    }));
+  };
+
+  resetList = () => {
+    this.setState({
+      goods: goodsFromServer,
+      inputValue: 1,
+      lastFunction: '',
+    });
+  };
+
+  repeatLastFunction = (name: string) => {
+    switch (name) {
+      case 'reverse':
+        this.reverseList();
+        break;
+      case 'sortAlpha':
+        this.sortAlphabetically();
+        break;
+      case 'sortByLength':
+        this.sortByLength();
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  render() {
+    const { goods, listIsVisible } = this.state;
+
```

Comment:
```suggestion
    const { goods, listIsVisible } = this.state;
    
    const goodsToRender = this.getGoodsToRender();
```
И goodsToRender используй как массив, который ты будешь мапить, чтобы получить список